contrib/remote_hsmd/hsmd.c

```diff
@@ -444,41 +515,45 @@ static struct io_plan *init_hsm(struct io_conn *conn,
 	 * will use */
 	c->chainparams = chainparams;
 
-	/* To support integration tests we honor any seed provided
-	 * in the hsm_secret file (testnet only). Otherwise we
-	 * generate a random seed.
-	 */
-	if (!read_test_seed(&hsm_secret)) {
-		randombytes_buf(&hsm_secret, sizeof(hsm_secret));
-	}
-
 	/* Is this a warm start (restart) or a cold start (first time)? */
-	coldstart = access("WARM", F_OK) == -1;
-
-	proxy_stat rv = proxy_init_hsm(&bip32_key_version, chainparams,
-				       coldstart, &hsm_secret,
-				       &node_id, &pubstuff.bip32);
-	if (PROXY_PERMANENT(rv)) {
-		status_failed(STATUS_FAIL_INTERNAL_ERROR,
-		              "proxy_%s failed: %s", __FUNCTION__,
-			      proxy_last_message());
-	}
-	else if (!PROXY_SUCCESS(rv)) {
-		status_unusual("proxy_%s failed: %s", __FUNCTION__,
-			       proxy_last_message());
-		return bad_req_fmt(conn, c, msg_in,
-				   "proxy_%s error: %s", __FUNCTION__,
-				   proxy_last_message());
-	}
+	if (restore_pubstuff(&node_id, &pubstuff.bip32, &bolt12)) {
+		// This is a warm start.
+		proxy_set_node_id(&node_id);
+	} else {
+		// This is a cold start, initialize the remote signer.
```

Comment:
Let's log something here, in case the operator messed up and didn't really mean to cold-start?

Reply:
done